lib/logflare/backends/adaptor/postgres_adaptor/pg_repo.ex

```diff
@@ -198,4 +173,93 @@ defmodule Logflare.Backends.Adaptor.PostgresAdaptor.PgRepo do
 
     repo.insert(changeset)
   end
+
+  ## Genserver calls
+
+  def start_link(%{repo_module_name: repo_module_name} = state),
+    do: GenServer.start_link(__MODULE__, state, name: :"PgRepo#{repo_module_name}")
```

Comment:
Could we  store the source backend config for the repo in state? It would result in less args being passed around

Reply:
Also, we can register different name using :via without needing to use another atom.

Reply:
can do 👍 decided to just send everything "organised" but can do that

and will do the change to use via 👍